beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -2051,4 +2051,149 @@ mod tests {
         assert!(connected_peers.contains(&peers[6]));
         assert!(connected_peers.contains(&peers[7]));
     }
+
+    // Test properties PeerManager should have using randomly generated input.
+    #[cfg(test)]
+    mod property_based_tests {
+        use crate::peer_manager::config::DEFAULT_TARGET_PEERS;
+        use crate::peer_manager::tests::build_peer_manager;
+        use crate::rpc::MetaData;
+        use libp2p::PeerId;
+        use quickcheck::{Arbitrary, Gen, TestResult};
+        use quickcheck_macros::quickcheck;
+        use tokio::runtime::Runtime;
+        use types::Unsigned;
+        use types::{EthSpec, MainnetEthSpec as E};
+
+        #[derive(Clone, Debug)]
+        struct PeerCondition {
+            direction: Direction,
+            attestation_subnets: Vec<AttestationSubnetId>,
+            sync_committee_subnets: Vec<SyncCommitteeSubnetId>,
+            score: f64,
+            gossipsub_score: f64,
+        }
+
+        impl Arbitrary for PeerCondition {
+            fn arbitrary(g: &mut Gen) -> Self {
+                PeerCondition {
+                    direction: Direction::arbitrary(g),
+                    attestation_subnets: Vec::arbitrary(g),
+                    sync_committee_subnets: Vec::arbitrary(g),
+                    score: f64::arbitrary(g),
+                    gossipsub_score: f64::arbitrary(g),
+                }
+            }
+        }
+
+        #[derive(Clone, Debug)]
+        enum Direction {
+            Outgoing,
+            Incoming,
+        }
+
+        impl Arbitrary for Direction {
+            fn arbitrary(g: &mut Gen) -> Self {
+                g.choose(&[Direction::Outgoing, Direction::Incoming])
+                    .unwrap()
+                    .clone()
+            }
+        }
+
+        #[derive(Clone, Debug)]
+        struct AttestationSubnetId(usize);
+
+        impl Arbitrary for AttestationSubnetId {
+            fn arbitrary(g: &mut Gen) -> Self {
+                let len = <E as EthSpec>::SubnetBitfieldLength::to_usize();
+                let id_range = (0..len).collect::<Vec<usize>>();
+                AttestationSubnetId(*g.choose(&id_range).unwrap())
+            }
+        }
+
+        #[derive(Clone, Debug)]
+        struct SyncCommitteeSubnetId(usize);
+
+        impl Arbitrary for SyncCommitteeSubnetId {
+            fn arbitrary(g: &mut Gen) -> Self {
+                let len = <E as EthSpec>::SyncCommitteeSubnetCount::to_usize();
+                let id_range = (0..len).collect::<Vec<usize>>();
+                SyncCommitteeSubnetId(*g.choose(&id_range).unwrap())
+            }
+        }
+
+        #[quickcheck]
+        fn prune_excess_peers(peer_conditions: Vec<PeerCondition>) -> TestResult {
+            let target_peer_count = DEFAULT_TARGET_PEERS;
+            let rt = Runtime::new().unwrap();
+
+            rt.block_on(async move {
```

Comment:
Creating `Runtime` instead of using `#[tokio::test]`, because ...

- `#[quickcheck]` and `#[tokio::test]` cannot declare at the same place. 
- have tried https://github.com/nytopop/quickcheck_async but it didn't work.

Reply:
Sure, this seems fine to me